src/pytorch_lightning/trainer/connectors/data_connector.py

```diff
@@ -580,3 +591,18 @@ def get_instance(self, hook_name: str) -> Union["pl.LightningModule", "pl.Lightn
                 " `LightningDataModule`. It will use the implementation from `LightningModule` instance."
             )
         return self.model
+
+
+def _check_dataloader_none(
+    dataloader: Optional[Union[TRAIN_DATALOADERS, EVAL_DATALOADERS]],
+    dataloader_source: _DataLoaderSource,
+    trainer_fn: TrainerFn,
+) -> None:
+    # A prefix in the message to disambiguate between the train- and (optional) val dataloader that .fit() accepts
+    prefix = "training " if trainer_fn == TrainerFn.FITTING else ""
+    if dataloader is None and not dataloader_source.is_defined():
+        raise ValueError(
+            f"No valid {prefix}dataloader was passed to `Trainer.{trainer_fn}()`."
```

Comment:
The message would be "No valid trainingdataloader ..."

Reply:
The prefix has a space :)